src/client/sandbox/event/drag-and-drop/data-transfer-item-list.js

```diff
@@ -0,0 +1,149 @@
+import DataTransferItem from './data-transfer-item';
+import DATA_STORE_MODE from './data-store-mode';
+import DATA_TRANSFER_ITEM_KIND from './data-transfer-item-kind';
+
+function parseTextUriList (textUriList) {
+    textUriList = textUriList.replace(/\r\n$/, '');
+
+    var res = [];
+
+    if (textUriList === '')
+        return res;
+
+    textUriList = textUriList.split(/\r\n/);
+
+    for (var i = 0; i < textUriList.length; i++) {
+        if (textUriList[0] !== '#')
+            res.push(textUriList[i]);
+    }
+
+    return res;
+}
+
+// https://html.spec.whatwg.org/multipage/interaction.html#datatransferitemlist
+export default class DataTransferItemList {
```

Comment:
In the native browser class wrappers we attempt to hide our internal properties and methods. And define public members through `defineProperty`. For example, this class should not has `_items` and `_dataStore` fields.